data_objects/lib/data_objects/ext/do_common.c

```diff
@@ -210,18 +210,20 @@ VALUE data_objects_parse_time(const char *date) {
     return Qnil;
   }
 
+  // TODO: support the timezone being returned from 
   return rb_funcall(rb_cTime, rb_intern("local"), 7, INT2NUM(year), INT2NUM(month), INT2NUM(day), INT2NUM(hour), INT2NUM(min), INT2NUM(sec), INT2NUM(usec));
```

Comment:
Yeah, we'd need to handle the cases different here then. Use `local` when there's no UTC offset present, but build an object with the specific UTC offset otherwise.